openrr-command/src/robot_command.rs

```diff
@@ -328,11 +328,13 @@ impl RobotCommandExecutor {
                 x,
                 y,
                 theta,
-                wait_secs,
+                duration_secs,
             } => {
-                client.send_velocity(&BaseVelocity::new(*x, *y, *theta))?;
-                // wait publish
-                sleep(Duration::from_secs_f64(*wait_secs));
+                let start = SystemTime::now();
+                while SystemTime::now() < (start + Duration::from_secs_f64(*duration_secs)) {
```

Comment:
It's better to use `Instant` (and `elapsed()`) in this case. Instant is monotonic clock.

https://doc.rust-lang.org/std/time/struct.Instant.html

`SystemTime` can be go back to the past.